[ICE]: Failed to normalize Alias(Free

<!--
[31mICE[0m: Rustc ./3BB5EEF5533B56D7EE8DD54D73D5A68854A47436446FD333FF46FF1D6E16B536.rs '-Zcrate-attr=feature(lazy_type_alias) -Znext-solver=globally -ooutputfile -Zdump-mir-dir=dir' 'error: internal compiler error: compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13: Failed to normalize Alias(Free, AliasTy { args: [{type error}], def_id: DefId(0:10 ~ 3BB5EEF5533B56D7EE8DD54D73D5A68854A47436446FD333FF46FF1D6E16B536[a85d]::Arr2), .. }) in typing_env=TypingEnv { typing_mode: PostAnalysis, param_env: ParamEnv { caller_bounds: [] } }, maybe try to call `try_normalize_erasing_regions` instead', 'error: internal compiler error: compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13: Failed to normalize Alias(Free, AliasTy { args: [{type error}], def_id: DefId(0:10 ~ 3BB5EEF5533B56D7EE8DD54D73D5A68854A47436446FD333FF46FF1D6E16B536[a85d]::Arr2), .. }) in typing_env=TypingEnv { typing_mode: PostAnalysis, param_env: ParamEnv { caller_bounds: [] } }, maybe try to call `try_normalize_erasing_regions` instead'
File: /tmp/im/3BB5EEF5533B56D7EE8DD54D73D5A68854A47436446FD333FF46FF1D6E16B536.rs
-->

auto-reduced (treereduce-rust):
````rust
//@compile-flags: -Znext-solver=globally
#![feature(lazy_type_alias)]
#![feature(min_generic_const_args)]

trait Trait {
    type const ASSOC: usize;
}

type Arr2<T> = [(); {
    {
        {
            <Arr2 as Trait>::ASSOC
        }
    }
}];

struct Default<
    const N: usize,
    const M: usize = {
        {
            {
                <T as Trait>::ASSOC
            }
        }
    },
>;

````

original:
````rust
//@ check-pass

#![feature(min_generic_const_args)]
#![expect(incomplete_features)]

struct Foo<const N: usize>;

trait Trait {
    type const ASSOC: usize;
}

type Arr<const N: usize> = [(); {{{ N }}}];
type Arr2<T> = [(); {{{ <Arr2 as Trait>::ASSOC }}}];
type Ty<const N: usize> = Foo<{{{ N }}}>;
type Ty2<T> = Foo<{{{ <T as Trait>::ASSOC }}}>;
struct Default<const N: usize, const M: usize = {{{ <T as Trait>::ASSOC }}}>;
struct Default2<T, const N: usize>(T);

fn repeat<T: Trait, const N: usize>() {
    let _1 = [(); {{{ N }}}];
    let _2 = [(); {{{ <T as Trait>::ASSOC }}}];
}

fn main() {}

````

Version information
````
rustc 1.95.0-nightly (a33907a7a 2026-02-14)
binary: rustc
commit-hash: a33907a7a5381473eec8bcfa0c56e05a856a911c
commit-date: 2026-02-14
host: x86_64-unknown-linux-gnu
release: 1.95.0-nightly
LLVM version: 22.1.0

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/a33907a7a5381473eec8bcfa0c56e05a856a911c/compiler/rustc_middle/src/ty/normalize_erasing_regions.rs#L165-L177

Command:
`/home/matthias/.rustup/toolchains/master/bin/rustc -Zcrate-attr=feature(lazy_type_alias) -Znext-solver=globally`

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0425]: cannot find type `T` in this scope
  --> /tmp/icemaker_global_tempdir.xNVTExHsALgM/rustc_testrunner_tmpdir_reporting.M3N7BWYS5Inc/mvce.rs:20:18
   |
20 |                 <T as Trait>::ASSOC
   |                  ^ not found in this scope

warning: the feature `min_generic_const_args` is incomplete and may not be safe to use and/or cause compiler crashes
 --> /tmp/icemaker_global_tempdir.xNVTExHsALgM/rustc_testrunner_tmpdir_reporting.M3N7BWYS5Inc/mvce.rs:1:12
  |
1 | #![feature(min_generic_const_args)]
  |            ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #132980 <https://github.com/rust-lang/rust/issues/132980> for more information
  = note: `#[warn(incomplete_features)]` on by default

warning: the feature `lazy_type_alias` is incomplete and may not be safe to use and/or cause compiler crashes
 --> <crate attribute>:1:12
  |
1 | #![feature(lazy_type_alias)]
  |            ^^^^^^^^^^^^^^^
  |
  = note: see issue #112792 <https://github.com/rust-lang/rust/issues/112792> for more information

error[E0601]: `main` function not found in crate `mvce`
  --> /tmp/icemaker_global_tempdir.xNVTExHsALgM/rustc_testrunner_tmpdir_reporting.M3N7BWYS5Inc/mvce.rs:24:3
   |
24 | >;
   |   ^ consider adding a `main` function to `/tmp/icemaker_global_tempdir.xNVTExHsALgM/rustc_testrunner_tmpdir_reporting.M3N7BWYS5Inc/mvce.rs`

error[E0107]: missing generics for type alias `Arr2`
  --> /tmp/icemaker_global_tempdir.xNVTExHsALgM/rustc_testrunner_tmpdir_reporting.M3N7BWYS5Inc/mvce.rs:10:14
   |
10 |             <Arr2 as Trait>::ASSOC
   |              ^^^^ expected 1 generic argument
   |
note: type alias defined here, with 1 generic parameter: `T`
  --> /tmp/icemaker_global_tempdir.xNVTExHsALgM/rustc_testrunner_tmpdir_reporting.M3N7BWYS5Inc/mvce.rs:7:6
   |
 7 | type Arr2<T> = [(); {
   |      ^^^^ -
help: add missing generic argument
   |
10 |             <Arr2<T> as Trait>::ASSOC
   |                  +++

error: internal compiler error: /rustc-dev/a33907a7a5381473eec8bcfa0c56e05a856a911c/compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13: Failed to normalize Alias(Free, AliasTy { args: [{type error}], def_id: DefId(0:5 ~ mvce[a81d]::Arr2), .. }) in typing_env=TypingEnv { typing_mode: PostAnalysis, param_env: ParamEnv { caller_bounds: [] } }, maybe try to call `try_normalize_erasing_regions` instead


thread 'rustc' (993037) panicked at /rustc-dev/a33907a7a5381473eec8bcfa0c56e05a856a911c/compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13:
Box<dyn Any>
stack backtrace:
   0:     0x7fac8f1cbb8b - <<std[3d5b93a78549aa0a]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[154cb5f5717d2f57]::fmt::Display>::fmt
   1:     0x7fac8f811788 - core[154cb5f5717d2f57]::fmt::write
   2:     0x7fac8f1e2e06 - <std[3d5b93a78549aa0a]::sys::stdio::unix::Stderr as std[3d5b93a78549aa0a]::io::Write>::write_fmt
   3:     0x7fac8f1a1c38 - std[3d5b93a78549aa0a]::panicking::default_hook::{closure#0}
   4:     0x7fac8f1beff3 - std[3d5b93a78549aa0a]::panicking::default_hook
   5:     0x7fac8e18fd8f - std[3d5b93a78549aa0a]::panicking::update_hook::<alloc[8f8f4e5073486fe0]::boxed::Box<rustc_driver_impl[564c6a6441c9b920]::install_ice_hook::{closure#1}>>::{closure#0}
   6:     0x7fac8f1bf2d2 - std[3d5b93a78549aa0a]::panicking::panic_with_hook
   7:     0x7fac8e1c3641 - std[3d5b93a78549aa0a]::panicking::begin_panic::<rustc_errors[1e193ea20fde428e]::ExplicitBug>::{closure#0}
   8:     0x7fac8e1bc3e6 - std[3d5b93a78549aa0a]::sys::backtrace::__rust_end_short_backtrace::<std[3d5b93a78549aa0a]::panicking::begin_panic<rustc_errors[1e193ea20fde428e]::ExplicitBug>::{closure#0}, !>
   9:     0x7fac8e1bc25b - std[3d5b93a78549aa0a]::panicking::begin_panic::<rustc_errors[1e193ea20fde428e]::ExplicitBug>
  10:     0x7fac8e1cebd1 - <rustc_errors[1e193ea20fde428e]::diagnostic::BugAbort as rustc_errors[1e193ea20fde428e]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
  11:     0x7fac8e797c49 - rustc_middle[573f590e3f45eafe]::util::bug::opt_span_bug_fmt::<rustc_span[7cfaf5af3f4ab127]::span_encoding::Span>::{closure#0}
  12:     0x7fac8e797dd2 - rustc_middle[573f590e3f45eafe]::ty::context::tls::with_opt::<rustc_middle[573f590e3f45eafe]::util::bug::opt_span_bug_fmt<rustc_span[7cfaf5af3f4ab127]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  13:     0x7fac8e78915b - rustc_middle[573f590e3f45eafe]::ty::context::tls::with_context_opt::<rustc_middle[573f590e3f45eafe]::ty::context::tls::with_opt<rustc_middle[573f590e3f45eafe]::util::bug::opt_span_bug_fmt<rustc_span[7cfaf5af3f4ab127]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  14:     0x7fac8be06f64 - rustc_middle[573f590e3f45eafe]::util::bug::bug_fmt
  15:     0x7fac8fc661bf - <rustc_middle[573f590e3f45eafe]::ty::normalize_erasing_regions::NormalizeAfterErasingRegionsFolder as rustc_type_ir[19d70f16965cce5f]::fold::TypeFolder<rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt>>::fold_ty
  16:     0x7fac8fc9e785 - rustc_ty_utils[bc12b725221b52f3]::instance::resolve_instance_raw
  17:     0x7fac8fc9b391 - rustc_query_impl[ff7f0a380a6f925b]::query_impl::resolve_instance_raw::invoke_provider_fn::__rust_begin_short_backtrace
  18:     0x7fac8fc9a660 - rustc_query_impl[ff7f0a380a6f925b]::execution::try_execute_query::<rustc_middle[573f590e3f45eafe]::query::caches::DefaultCache<rustc_middle[573f590e3f45eafe]::ty::PseudoCanonicalInput<(rustc_span[7cfaf5af3f4ab127]::def_id::DefId, &rustc_middle[573f590e3f45eafe]::ty::list::RawList<(), rustc_middle[573f590e3f45eafe]::ty::generic_args::GenericArg>)>, rustc_middle[573f590e3f45eafe]::query::erase::ErasedData<[u8; 32usize]>>, {rustc_query_impl[ff7f0a380a6f925b]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}, false>
  19:     0x7fac8fc9a2a9 - rustc_query_impl[ff7f0a380a6f925b]::query_impl::resolve_instance_raw::get_query_non_incr::__rust_end_short_backtrace
  20:     0x7fac8fae227f - <rustc_middle[573f590e3f45eafe]::ty::instance::Instance>::try_resolve
  21:     0x7fac8c4c4ad2 - <rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt>::const_eval_resolve_for_typeck
  22:     0x7fac8fb6c7da - rustc_trait_selection[284017491e4b25b4]::traits::try_evaluate_const
  23:     0x7fac90e45c95 - <rustc_trait_selection[284017491e4b25b4]::solve::delegate::SolverDelegate as rustc_next_trait_solver[a24d63ab66b75394]::delegate::SolverDelegate>::evaluate_const
  24:     0x7fac8ef6fe3a - <rustc_next_trait_solver[a24d63ab66b75394]::solve::eval_ctxt::EvalCtxt<rustc_trait_selection[284017491e4b25b4]::solve::delegate::SolverDelegate, rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt>>::compute_const_evaluatable_goal
  25:     0x7fac90aa0339 - <rustc_next_trait_solver[a24d63ab66b75394]::solve::search_graph::SearchGraphDelegate<rustc_trait_selection[284017491e4b25b4]::solve::delegate::SolverDelegate> as rustc_type_ir[19d70f16965cce5f]::search_graph::Delegate>::compute_goal::{closure#0}
  26:     0x7fac90a9c117 - <rustc_type_ir[19d70f16965cce5f]::search_graph::SearchGraph<rustc_next_trait_solver[a24d63ab66b75394]::solve::search_graph::SearchGraphDelegate<rustc_trait_selection[284017491e4b25b4]::solve::delegate::SolverDelegate>, rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt>>::evaluate_goal_in_task
  27:     0x7fac90a9bc11 - rustc_trait_selection[284017491e4b25b4]::solve::evaluate_root_goal_for_proof_tree_raw
  28:     0x7fac90a9ba5c - rustc_query_impl[ff7f0a380a6f925b]::query_impl::evaluate_root_goal_for_proof_tree_raw::invoke_provider_fn::__rust_begin_short_backtrace
  29:     0x7fac9035522f - rustc_query_impl[ff7f0a380a6f925b]::execution::try_execute_query::<rustc_middle[573f590e3f45eafe]::query::caches::DefaultCache<rustc_type_ir[19d70f16965cce5f]::canonical::CanonicalQueryInput<rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt, rustc_type_ir[19d70f16965cce5f]::solve::QueryInput<rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt, rustc_middle[573f590e3f45eafe]::ty::predicate::Predicate>>, rustc_middle[573f590e3f45eafe]::query::erase::ErasedData<[u8; 48usize]>>, {rustc_query_impl[ff7f0a380a6f925b]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}, false>
  30:     0x7fac90354e87 - rustc_query_impl[ff7f0a380a6f925b]::query_impl::evaluate_root_goal_for_proof_tree_raw::get_query_non_incr::__rust_end_short_backtrace
  31:     0x7fac90354d99 - <rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt as rustc_type_ir[19d70f16965cce5f]::interner::Interner>::evaluate_root_goal_for_proof_tree_raw
  32:     0x7fac907b25ea - <rustc_trait_selection[284017491e4b25b4]::solve::delegate::SolverDelegate as rustc_next_trait_solver[a24d63ab66b75394]::solve::eval_ctxt::SolverDelegateEvalExt>::evaluate_root_goal_for_proof_tree
  33:     0x7fac907b0e8c - <rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::InspectCandidate>::instantiate_proof_tree_for_nested_goal
  34:     0x7fac907b5832 - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  35:     0x7fac907b4230 - <rustc_infer[ce7fcaac309b9266]::infer::InferCtxt as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::InferCtxtProofTreeExt>::visit_proof_tree_at_depth::<rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation>
  36:     0x7fac907b49aa - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  37:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  38:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  39:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  40:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  41:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  42:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  43:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  44:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  45:     0x7fac907b477f - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  46:     0x7fac907b4230 - <rustc_infer[ce7fcaac309b9266]::infer::InferCtxt as rustc_trait_selection[284017491e4b25b4]::solve::inspect::analyse::InferCtxtProofTreeExt>::visit_proof_tree_at_depth::<rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::BestObligation>
  47:     0x7fac907b80cf - rustc_trait_selection[284017491e4b25b4]::solve::fulfill::derive_errors::find_best_leaf_obligation
  48:     0x7fac907b772b - <rustc_trait_selection[284017491e4b25b4]::traits::FulfillmentError as rustc_infer[ce7fcaac309b9266]::traits::engine::FromSolverError<rustc_trait_selection[284017491e4b25b4]::solve::fulfill::NextSolverError>>::from_solver_error
  49:     0x7fac90337a19 - <rustc_trait_selection[284017491e4b25b4]::solve::fulfill::FulfillmentCtxt<rustc_trait_selection[284017491e4b25b4]::traits::FulfillmentError> as rustc_infer[ce7fcaac309b9266]::traits::engine::TraitEngine<rustc_trait_selection[284017491e4b25b4]::traits::FulfillmentError>>::try_evaluate_obligations
  50:     0x7fac9028734d - <rustc_trait_selection[284017491e4b25b4]::solve::normalize::NormalizationFolder<rustc_trait_selection[284017491e4b25b4]::traits::FulfillmentError>>::normalize_alias_term
  51:     0x7fac90d085b9 - <rustc_trait_selection[284017491e4b25b4]::solve::normalize::NormalizationFolder<rustc_trait_selection[284017491e4b25b4]::traits::FulfillmentError> as rustc_type_ir[19d70f16965cce5f]::fold::FallibleTypeFolder<rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt>>::try_fold_const::{closure#0}
  52:     0x7fac90288017 - <rustc_middle[573f590e3f45eafe]::ty::Ty as rustc_type_ir[19d70f16965cce5f]::fold::TypeSuperFoldable<rustc_middle[573f590e3f45eafe]::ty::context::TyCtxt>>::try_super_fold_with::<rustc_trait_selection[284017491e4b25b4]::solve::normalize::NormalizationFolder<rustc_trait_selection[284017491e4b25b4]::traits::FulfillmentError>>
  53:     0x7fac9029b48d - <rustc_hir_analysis[79dc58a3f763e39a]::check::wfcheck::WfCheckingCtxt>::deeply_normalize::<rustc_middle[573f590e3f45eafe]::ty::Ty>
  54:     0x7fac90064ee7 - rustc_hir_analysis[79dc58a3f763e39a]::check::check::check_item_type
  55:     0x7fac900578ee - rustc_hir_analysis[79dc58a3f763e39a]::check::wfcheck::check_well_formed
  56:     0x7fac900578cf - rustc_query_impl[ff7f0a380a6f925b]::query_impl::check_well_formed::invoke_provider_fn::__rust_begin_short_backtrace
  57:     0x7fac900570c6 - rustc_query_impl[ff7f0a380a6f925b]::execution::try_execute_query::<rustc_data_structures[52ca772bd10c471]::vec_cache::VecCache<rustc_span[7cfaf5af3f4ab127]::def_id::LocalDefId, rustc_middle[573f590e3f45eafe]::query::erase::ErasedData<[u8; 1usize]>, rustc_middle[573f590e3f45eafe]::dep_graph::graph::DepNodeIndex>, {rustc_query_impl[ff7f0a380a6f925b]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}, false>
  58:     0x7fac90056e42 - rustc_query_impl[ff7f0a380a6f925b]::query_impl::check_well_formed::get_query_non_incr::__rust_end_short_backtrace
  59:     0x7fac900543a9 - rustc_hir_analysis[79dc58a3f763e39a]::check::wfcheck::check_type_wf
  60:     0x7fac900542a9 - rustc_query_impl[ff7f0a380a6f925b]::query_impl::check_type_wf::invoke_provider_fn::__rust_begin_short_backtrace
  61:     0x7fac90a48fd3 - rustc_query_impl[ff7f0a380a6f925b]::execution::try_execute_query::<rustc_middle[573f590e3f45eafe]::query::caches::SingleCache<rustc_middle[573f590e3f45eafe]::query::erase::ErasedData<[u8; 1usize]>>, {rustc_query_impl[ff7f0a380a6f925b]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}, false>
  62:     0x7fac90a48d8e - rustc_query_impl[ff7f0a380a6f925b]::query_impl::check_type_wf::get_query_non_incr::__rust_end_short_backtrace
  63:     0x7fac901cb173 - rustc_hir_analysis[79dc58a3f763e39a]::check_crate
  64:     0x7fac901cf213 - rustc_interface[19fbbc036bde6456]::passes::analysis
  65:     0x7fac90a4a899 - rustc_query_impl[ff7f0a380a6f925b]::execution::try_execute_query::<rustc_middle[573f590e3f45eafe]::query::caches::SingleCache<rustc_middle[573f590e3f45eafe]::query::erase::ErasedData<[u8; 0usize]>>, {rustc_query_impl[ff7f0a380a6f925b]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}, false>
  66:     0x7fac90a4a5d4 - rustc_query_impl[ff7f0a380a6f925b]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  67:     0x7fac90b54b67 - <rustc_interface[19fbbc036bde6456]::passes::create_and_enter_global_ctxt<core[154cb5f5717d2f57]::option::Option<rustc_interface[19fbbc036bde6456]::queries::Linker>, rustc_driver_impl[564c6a6441c9b920]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[154cb5f5717d2f57]::ops::function::FnOnce<(&rustc_session[2a7b4f38788f93d2]::session::Session, rustc_middle[573f590e3f45eafe]::ty::context::CurrentGcx, alloc[8f8f4e5073486fe0]::sync::Arc<rustc_data_structures[52ca772bd10c471]::jobserver::Proxy>, &std[3d5b93a78549aa0a]::sync::once_lock::OnceLock<rustc_middle[573f590e3f45eafe]::ty::context::GlobalCtxt>, &rustc_data_structures[52ca772bd10c471]::sync::worker_local::WorkerLocal<rustc_middle[573f590e3f45eafe]::arena::Arena>, &rustc_data_structures[52ca772bd10c471]::sync::worker_local::WorkerLocal<rustc_hir[1d49f14d19e2e96b]::Arena>, rustc_driver_impl[564c6a6441c9b920]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
  68:     0x7fac909f43b7 - rustc_interface[19fbbc036bde6456]::interface::run_compiler::<(), rustc_driver_impl[564c6a6441c9b920]::run_compiler::{closure#0}>::{closure#1}
  69:     0x7fac90a34bbe - std[3d5b93a78549aa0a]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[19fbbc036bde6456]::util::run_in_thread_with_globals<rustc_interface[19fbbc036bde6456]::util::run_in_thread_pool_with_globals<rustc_interface[19fbbc036bde6456]::interface::run_compiler<(), rustc_driver_impl[564c6a6441c9b920]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  70:     0x7fac90a352a0 - <std[3d5b93a78549aa0a]::thread::lifecycle::spawn_unchecked<rustc_interface[19fbbc036bde6456]::util::run_in_thread_with_globals<rustc_interface[19fbbc036bde6456]::util::run_in_thread_pool_with_globals<rustc_interface[19fbbc036bde6456]::interface::run_compiler<(), rustc_driver_impl[564c6a6441c9b920]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[154cb5f5717d2f57]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  71:     0x7fac90a3612c - <std[3d5b93a78549aa0a]::sys::thread::unix::Thread>::new::thread_start
  72:     0x7fac8a4a598b - <unknown>
  73:     0x7fac8a529a0c - <unknown>
  74:                0x0 - <unknown>

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.95.0-nightly (a33907a7a 2026-02-14) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z crate-attr=feature(lazy_type_alias) -Z next-solver=globally -Z dump-mir-dir=dir

query stack during panic:
#0 [resolve_instance_raw] resolving instance `<Arr2<{type error}> as Trait>::ASSOC`
#1 [evaluate_root_goal_for_proof_tree_raw] computing proof tree for `the constant `<Arr2<{type error}> as Trait>::ASSOC` can be evaluated`
#2 [check_well_formed] checking that `Arr2` is well-formed
#3 [check_type_wf] checking that types are well-formed
#4 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 4 previous errors; 2 warnings emitted

Some errors have detailed explanations: E0107, E0425, E0601.
For more information about an error, try `rustc --explain E0107`.

```

</p>
</details>

<!--
query stack:
error: internal compiler error: /rustc-dev/a33907a7a5381473eec8bcfa0c56e05a856a911c/compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13: Failed to normalize Alias(Free, AliasTy { args: [{type error}], def_id: DefId(0:5 ~ mvce[a81d]::Arr2), .. }) in typing_env=TypingEnv { typing_mode: PostAnalysis, param_env: ParamEnv { caller_bounds: [] } }, maybe try to call `try_normalize_erasing_regions` instead
#0 [resolve_instance_raw] resolving instance `<Arr2<{type error}> as Trait>::ASSOC`
#1 [evaluate_root_goal_for_proof_tree_raw] computing proof tree for `the constant `<Arr2<{type error}> as Trait>::ASSOC` can be evaluated`
#2 [check_well_formed] checking that `Arr2` is well-formed
#3 [check_type_wf] checking that types are well-formed
#4 [analysis] running analysis passes on crate `mvce`

-->
@rustbot label +F-min_generic_const_args +F-lazy_type_alias +WG-trait-system-refactor

--- a/tests/ui/type-alias/recursive-lazy-type-alias-ice-152633.rs
+++ b/tests/ui/type-alias/recursive-lazy-type-alias-ice-152633.rs
@@ -1,0 +1,14 @@
+//! Ensure a self-referencing lazy type alias with `min_generic_const_args`
+//! doesn't ICE during normalization.
+//!
+//! Regression test for <https://github.com/rust-lang/rust/issues/152633>.
+
+#![feature(lazy_type_alias)]
+#![feature(min_generic_const_args)]
+
+trait Trait {
+    type const ASSOC: ();
+}
+type Arr2 = [usize; <Arr2 as Trait>::ASSOC]; //~ ERROR E0275
+
+fn main() {}

--- a/tests/ui/type-alias/recursive-lazy-type-alias-ice-152633.stderr
+++ b/tests/ui/type-alias/recursive-lazy-type-alias-ice-152633.stderr
@@ -1,0 +1,11 @@
+error[E0275]: overflow normalizing the type alias `Arr2`
+  --> $DIR/recursive-lazy-type-alias-ice-152633.rs:12:1
+   |
+LL | type Arr2 = [usize; <Arr2 as Trait>::ASSOC];
+   | ^^^^^^^^^
+   |
+   = note: in case this is a recursive type alias, consider using a struct, enum, or union instead
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0275`.

